File: src/2015/1/a/js/solution.js
const { readInputData } = require('../../../../util');

const input = readInputData(__dirname);

const increment = '(';
const decrement = ')';

const steps = input.split('');


const destination = steps
  .reduce((floor, instruction) => {
    if (instruction === increment) {
      return floor + 1;
    }

    if (instruction === decrement) {
      return floor - 1;
    }

    return floor;
  }, 0);

console.log('Santa is on floor', destination);

File: src/2015/1/a/c/solution.c
#include "../../../../util/util.h"

#define INCREMENT '('
#define DECREMENT ')'


int main() {
  char buffer = 0;
  int floor = 0;

  while (mread(&buffer, 1) > 0) {
    if (buffer == INCREMENT) {
      floor += 1;
    }
    if (buffer == DECREMENT) {
      floor -= 1;
    }
  }

  mprintf("Santa is on floor %d\n", floor);

  return 0;
}
